site/content/integrate/slash-commands/_index.md

```diff
@@ -0,0 +1,129 @@
+```
+
+Be sure to set the `Content-Type` to `application/json` otherwise your response will be displayed as plaintext.
+
+### Parameters
+
+Slash command responses support more than just the `text` field. Here is a full list of supported parameters.
+
+| Parameter | Description | Required |
+|---|---|---|
+| text | [Markdown-formatted](https://docs.mattermost.com/help/messaging/formatting-text.html) message to display in the post. | If `attachments` is not set, yes |
+| response\_type | Set to `ephemeral` to reply with a message that only the user can see. <br> Set to blank or `in_channel` to create a regular message.<br> Defaults to `in_channel`. | No |
+| username | Overrides the username the message posts as.<br> Defaults to the username set during webhook creation or the webhook creator's username if the former was not set.<br> Must be enabled [in the configuration](https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-usernames). | No |
+| icon\_url | Overrides the profile picture the message posts with.<br> Defaults to the URL set during webhook creation or the webhook creator's profile picture if the former was not set.<br> Must be enabled [in the configuration](https://docs.mattermost.com/administration/config-settings.html#enable-integrations-to-override-profile-picture-icons). | No |
+| attachments | [Message attachments](https://docs.mattermost.com/developer/message-attachments.html) used for richer formatting options. | If `text` is not set, yes |
+| type | Sets the post `type`, mainly for use by plugins.<br> If not blank, must begin with `custom_`. Passing `attachments` will ignore this field and set the type to `slack_attachment`. | No |
+| props | Sets the post `props`, a JSON property bag for storing extra or meta data on the post.<br> Mainly used by other integrations accessing posts through the REST API.<br> The following keys are reserved: `from_webhook`, `override_username`, `override_icon_url` and `attachments`. | No |
+
+An example request using some more parameters would look like this:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Content-Length: 696
+
+{
+    "response_type": "in_channel",
+    "text": "\n#### Test results for July 27th, 2017\n@channel here are the requested test results.\n\n| Component  | Tests Run   | Tests Failed                                   |\n| ---------- | ----------- | ---------------------------------------------- |\n| Server     | 948         | :white_check_mark: 0                           |\n| Web Client | 123         | :warning: 2 [(see details)](http://linktologs) |\n| iOS Client | 78          | :warning: 3 [(see details)](http://linktologs) |\n\t\t      ",
+    "username": "test-automation"
+    "icon_url": "https://www.mattermost.org/wp-content/uploads/2016/04/icon.png",
+    "props": {
+        "test_data": {
+            "ios": 78,
+            "server": 948,
+            "web": 123
+        }
+    },
+}
+```
+
+### FAQ
```

Comment:
Are these worthwhile to include, or are they admin errors? https://docs.mattermost.com/developer/slash-commands.html#troubleshooting

Reply:
Neat! I'll integrate some of those comments into this document.